lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
+            returned flux light curve.
+        **kwargs : dict
+            Optional arguments to be passed to
+            `~lightkurve.correctors.PLDCorrector.create_pymc_model`,
+            `~lightkurve.correctors.PLDCorrector.optimize`, and
+            `~lightkurve.correctors.PLDCorrector.sample`.
+
+        Returns
+        -------
+        corrected_lc : `~lightkurve.lightcurve.LightCurve`
+            Systematics-corrected light curve.
+        """
+        model = self.create_pymc_model(**kwargs)
+        solution_or_trace = self.optimize(model=model, **kwargs)
+        if sample:
+            solution_or_trace = self.sample(model=model, start=solution_or_trace, **kwargs)
+
+        if remove_gp_trend:
+            variable = 'corrected_flux_without_gp'
         else:
-            # compute the coefficients C on the basis vectors;
-            # the PLD design matrix will be dotted with C to solve for the noise model.
-            ivar = 1.0 / M(rawflux_err)**2 # inverse variance
-            A = np.dot(MX.T, MX * ivar[:, None])
-            B = np.dot(MX.T, M(rawflux) * ivar)
-
-        # apply prior to design matrix weights for numerical stability
-        A[np.diag_indices_from(A)] += 1e-8
-        C = np.linalg.solve(A, B)
-
-        # compute detrended light curve
-        model = np.dot(X, C)
-        self.detrended_flux = rawflux - (model - np.nanmean(model))
-
-        # Create and return a new LightCurve object with the corrected flux
-        corrected_lc = lc.copy()[nanmask]
-        corrected_lc.flux = self.detrended_flux
-        corrected_lc.flux_err = flux_err
-        return corrected_lc
+            variable = 'corrected_flux'
+        return self._lightcurve_from_solution(solution_or_trace, variable=variable)
+
+    def _lightcurve_from_solution(self, solution_or_trace, variable='corrected_flux'):
+        """Helper function to generate light curve objects from the maximum
+        likelihood solution or the sample trace.
+
+        Parameters
+        ----------
+        solution_or_trace : dict or `~pymc3.backends.base.MultiTrace`
+            The output returned by the `optimize()` or `sample()` methods
+            of this object.
+        variable : str
+            Key for determining which light curve to extract from the given
+            solution or trace.
+
+        Returns
+        -------
+        lc : `~lightkurve.LightCurve`
+            Light curve object corresponding to the model variable.
+        """
+        lc = self.raw_lc.copy()
+        # If the model was sampled, use the mean and std of the flux values.
+        if isinstance(solution_or_trace, pm.backends.base.MultiTrace):
+            lc.flux = np.nanmean(solution_or_trace[variable], axis=0)
+            lc.flux_err = np.nanstd(solution_or_trace[variable], axis=0)
+        else:  # Otherwise, use the maximum likelihood flux values.
+            lc.flux = solution_or_trace[variable]
+        return lc
```

Comment:
Can you also change the `label` attribute to "PLD Corrected targetid", so that we can tell the difference? Also, you could add the motion model to the `meta` attribute, so that we can keep hold of it and use it down the line.